### PR TITLE
[Data] Update Ray Data documentation for {landing,overview,key concepts}

### PR DESCRIPTION
**Pull request by c21**
## Why are these changes needed?

This PR is to update Ray Data documentation for landing, overview and key concepts page, as discussed offline.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [x] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/data/overview.rst
Comment (scottjlee) [inline doc/source/data/overview.rst:39]:
```suggestion
    Ray Data is built on Ray, so it easily scales on a heterogeneous cluster, which has different types of CPU and GPU machines. Code that works on one machine also runs on a large cluster without any changes.
```

File: doc/source/data/overview.rst
Comment (scottjlee) [inline doc/source/data/overview.rst:41]:
```suggestion
    Ray Data can easily scale to hundreds of nodes to process hundreds of TB of data.
```

File: doc/source/data/overview.rst
Comment (scottjlee) [inline doc/source/data/overview.rst:45]:
```suggestion
    With Ray Data, you can express batch inference and ML training job directly under the same Ray Dataset API.
```

File: doc/source/ray-overview/getting-started.md
Comment (scottjlee) [inline doc/source/ray-overview/getting-started.md:19]: i think the actual title text should remain the same, but update the reference:
```suggestion
Scale offline inference and training ingest with [Ray Data](data_quickstart) --
```

**Review by scottjlee — APPROVED** [review]
LGTM, pending docs build failure (I can't view the built docs, might be worth doing a final check once build works)

File: doc/source/data/overview.rst
Comment (c21) [reply doc/source/data/overview.rst]: Updated.

File: doc/source/data/overview.rst
Comment (c21) [reply doc/source/data/overview.rst]: Updated.

File: doc/source/data/overview.rst
Comment (c21) [reply doc/source/data/overview.rst]: Updated.

File: doc/source/ray-overview/getting-started.md
Comment (c21) [reply doc/source/ray-overview/getting-started.md]: Updated.

File: doc/source/data/overview.rst
Comment (bveeramani) [inline doc/source/data/overview.rst:27]: Feel like "Through its powerful streaming execution" is redundant since we say "Ray Data streams" in the next clause
```suggestion
    Ray Data is designed for deep learning applications that involve both CPU preprocessing and GPU inference. Ray Data streams working data from CPU preprocessing tasks to GPU inferencing or training tasks, allowing you to utilize both sets of resources concurrently.
```

File: doc/source/data/overview.rst
Comment (bveeramani) [inline doc/source/data/overview.rst:45]: I know this is also in the draft for the blog post, but IMO "Python first" is more compelling than "Unified API and backend for batch inference and ML training."

Don't need to block on this since we can always revise later.

File: doc/source/data/overview.rst
Comment (bveeramani) [inline doc/source/data/overview.rst:37]:
```suggestion
.. dropdown:: Out-of-the-box scaling on heterogeneous clusters
```

**Review by bveeramani — APPROVED** [review]

**Review by omatthew98 — APPROVED** [review]
Looks good, think spotted one tiny typo.

File: doc/source/tune/tutorials/tune_get_data_in_and_out.md
Comment (omatthew98) [inline doc/source/tune/tutorials/tune_get_data_in_and_out.md:74]: Think it should be \[Dataset**s**\]\(data_quickstart\).

File: doc/source/data/overview.rst
Comment (c21) [reply doc/source/data/overview.rst]: Updated.

File: doc/source/data/overview.rst
Comment (c21) [reply doc/source/data/overview.rst]: Updated.

File: doc/source/tune/tutorials/tune_get_data_in_and_out.md
Comment (c21) [reply doc/source/tune/tutorials/tune_get_data_in_and_out.md]: Updated.

**Review by matthewdeng — APPROVED** [review]

**Pull request merged**
